### PR TITLE
[Data sync] Rewrite data and photo sync workers to be more robust and readable

### PR DESCRIPTION
**Pull request by gino-m**
1. Try to process entire queue each time worker is run rather than just mutation for a specific enqueued LOI. This makes the sync worker more robust in the case that a worker failed for a particular LOI before it could request a retry() or update the mutation status.
2. Introduces the `UploadQueueEntry` wrapper / grouping for mutations to be used by the workers and UI.

Towards #2840

### CODE REVIEW

**Comment by gino-m** [comment]
@shobhitagarwal1612 @anandwana001 PTAL?

File: ground/src/main/java/com/google/android/ground/persistence/sync/MediaUploadWorker.kt [hunk 1]
```diff
@@ -133,23 +94,10 @@ constructor(
...
-
-  companion object {
-    private const val LOI_ID = "locationOfInterestId"
-
-    @JvmStatic
-    fun createInputData(locationOfInterestId: String): Data =
-      Data.Builder().putString(LOI_ID, locationOfInterestId).build()
-
-    private fun SubmissionMutation.canRetry() =
-      this.retryCount < Config.MAX_MEDIA_UPLOAD_RETRY_COUNT
```
Comment (shobhitagarwal1612) [inline ground/src/main/java/com/google/android/ground/persistence/sync/MediaUploadWorker.kt:153]: Let's cleanup the unused constant as well from `Config.kt`

File: ground/src/main/java/com/google/android/ground/util/Debug.kt [hunk 2]
```diff
@@ -40,3 +42,5 @@ object Debug {
       null
     }
 }
+
+fun Throwable.priority() = if (this is FirebaseNetworkException) Log.DEBUG else Log.ERROR
```
Comment (shobhitagarwal1612) [inline ground/src/main/java/com/google/android/ground/util/Debug.kt:46]: How about adding a check [here](https://github.com/google/ground-android/blob/e2c703377ab69d81f755720693c326879b943bde/ground/src/main/java/com/google/android/ground/GroundApplication.kt#L63) and log to console?

Then we can continue to use `Timber.e` normally in all classes

File: ground/src/main/java/com/google/android/ground/persistence/sync/MediaUploadWorker.kt [hunk 3]
```diff
@@ -133,23 +94,10 @@ constructor(
...
       remoteStorageManager.uploadMediaFromFile(photoFile, path)
-      kotlin.Result.success(Unit)
-    } catch (t: FirebaseNetworkException) {
-      Timber.d(t, "Can't connect to Firebase to upload photo")
-      kotlin.Result.failure(t)
+      return kotlin.Result.success(Unit)
     } catch (t: Throwable) {
-      Timber.e(t, "Failed to upload photo")
-      kotlin.Result.failure(t)
+      Timber.log(t.priority(), t, "Photo upload failed")
```
Comment (shobhitagarwal1612) [inline ground/src/main/java/com/google/android/ground/persistence/sync/MediaUploadWorker.kt:99]: Please see the other comment regarding reverting this change

File: ground/src/main/java/com/google/android/ground/repository/MutationRepository.kt [hunk 4]
```diff
@@ -44,69 +45,95 @@ import kotlinx.coroutines.flow.combine
 class MutationRepository
 @Inject
 constructor(
-  private val localSurveyStore: LocalSurveyStore,
   private val localLocationOfInterestStore: LocalLocationOfInterestStore,
   private val localSubmissionStore: LocalSubmissionStore,
+  private val authenticationManager: AuthenticationManager,
```
Comment (shobhitagarwal1612) [inline ground/src/main/java/com/google/android/ground/repository/MutationRepository.kt:50]: Can we keep this list alphabetically sorted?

**Review by shobhitagarwal1612 — CHANGES_REQUESTED** [review]
I took a first pass and left some comments. Will take another pass soon.

File: ground/src/main/java/com/google/android/ground/model/submission/UploadQueueEntry.kt [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+import com.google.android.ground.model.mutation.LocationOfInterestMutation
+import com.google.android.ground.model.mutation.Mutation
+import com.google.android.ground.model.mutation.SubmissionMutation
+import java.util.Date
+
+/**
+ * A set of changes to be applied to the remote datastore, initiated by the user by completing the
+ * data collection flow and clicking "Submit". Each instance of [UploadQueueEntry] shows up as a
+ * separate item in the "Upload status" screen.
```
Comment (shobhitagarwal1612) [inline ground/src/main/java/com/google/android/ground/model/submission/UploadQueueEntry.kt:27]: > Each instance of [UploadQueueEntry] shows up as a  separate item in the "Upload status" screen.
 
 Is this also implemented in the same PR?

File: ground/src/main/java/com/google/android/ground/repository/MutationRepository.kt [hunk 6]
```diff
@@ -54,59 +55,85 @@ constructor(
...
+
+  private suspend fun buildUploadQueue(
+    loiMutations: List<LocationOfInterestMutation>,
+    submissionMutations: List<SubmissionMutation>,
+  ): List<UploadQueueEntry> {
+    val user = authenticationManager.getAuthenticatedUser()
+    val loiMutationMap = loiMutations.filterByUser(user).associateBy { it.collectionId }
+    val submissionMutationMap =
+      submissionMutations.filterByUser(user).associateBy { it.collectionId }
+    val collectionIds = loiMutationMap.keys + submissionMutationMap.keys
```
Comment (shobhitagarwal1612) [inline ground/src/main/java/com/google/android/ground/repository/MutationRepository.kt:111]: Do we need to remove duplicates?

**Review by shobhitagarwal1612 — APPROVED** [review]

File: ground/src/main/java/com/google/android/ground/persistence/sync/MediaUploadWorker.kt (hunk 1 above)
Comment (gino-m) [reply ground/src/main/java/com/google/android/ground/persistence/sync/MediaUploadWorker.kt:153]: Done!

File: ground/src/main/java/com/google/android/ground/repository/MutationRepository.kt (hunk 6 above)
Comment (gino-m) [reply ground/src/main/java/com/google/android/ground/repository/MutationRepository.kt:111]: `collectionIds` is a `Set, so no need. I've added an explicit type to the `val` to make this clear.

File: ground/src/main/java/com/google/android/ground/persistence/sync/MediaUploadWorker.kt (hunk 3 above)
Comment (gino-m) [reply ground/src/main/java/com/google/android/ground/persistence/sync/MediaUploadWorker.kt:99]: Forked this discussion into a suggestion in #2891.

**Pull request merged**